Formula/v8.rb

```diff
@@ -15,98 +17,217 @@ class V8 < Formula
+      :revision => "0653eee0c81ea04715c635dd0885e8096ff6ba6d"
+  end
+
+  resource "llvm--project-libcxx" do
+    url "https://chromium.googlesource.com/chromium/llvm-project/libcxx.git",
+      :revision => "f56f1bba1ade4a408d403ff050d50e837bae47df"
+  end
+
+  resource "llvm--project-libcxxabi" do
+    url "https://chromium.googlesource.com/chromium/llvm-project/libcxxabi.git",
+      :revision => "05ba3281482304ae8de31123a594972a495da06d"
+  end
+
+  resource "v8-deps-third_party-benchmarks" do
+    url "https://chromium.googlesource.com/v8/deps/third_party/benchmarks.git",
+      :revision => "05d7188267b4560491ff9155c5ee13e207ecd65f"
+  end
+
+  resource "v8-deps-third_party-mozilla--tests" do
+    url "https://chromium.googlesource.com/v8/deps/third_party/mozilla-tests.git",
+      :revision => "f6c578a10ea707b1a8ab0b88943fe5115ce2b9be"
+  end
+
+  # not building on Yosemite
+  # https://bugs.chromium.org/p/chromium/issues/detail?id=620127
+  depends_on :macos => :el_capitan
+
+  # depot_tools/GN require Python 2.7+
+  depends_on "python@2" => :build
+
+  depends_on "readline" => :optional
+
+  needs :cxx11
+
   def install
-    # Bully GYP into correctly linking with c++11
-    ENV.cxx11
-    ENV["GYP_DEFINES"] = "clang=1 mac_deployment_target=#{MacOS.version}"
-    # https://code.google.com/p/v8/issues/detail?id=4511#c3
-    ENV.append "GYP_DEFINES", "v8_use_external_startup_data=0"
-
-    # fix up libv8.dylib install_name
-    # https://github.com/Homebrew/homebrew/issues/36571
-    # https://code.google.com/p/v8/issues/detail?id=3871
-    inreplace "tools/gyp/v8.gyp",
-              "'OTHER_LDFLAGS': ['-dynamiclib', '-all_load']",
-              "\\0, 'DYLIB_INSTALL_NAME_BASE': '#{opt_lib}'"
-
-    (buildpath/"build/gyp").install resource("gyp")
-    (buildpath/"third_party/icu").install resource("icu")
-    (buildpath/"buildtools").install resource("buildtools")
-    (buildpath/"base/trace_event/common").install resource("common")
-    (buildpath/"tools/swarming_client").install resource("swarming_client")
-    (buildpath/"testing/gtest").install resource("gtest")
-    (buildpath/"testing/gmock").install resource("gmock")
-    (buildpath/"tools/clang").install resource("clang")
-
-    system "make", "native", "library=shared", "snapshot=on",
-                   "console=readline", "i18nsupport=off",
-                   "strictaliasing=off"
-
-    include.install Dir["include/*"]
-
-    cd "out/native" do
-      rm ["libgmock.a", "libgtest.a"]
-      lib.install Dir["lib*"]
-      bin.install "d8", "mksnapshot", "process", "shell" => "v8"
+    v8_version = head? ? "HEAD" : version
+    arch = MacOS.prefer_64_bit? ? "x64" : "x86"
+    output_name = "#{arch}.release"
+    output_path = "out.gn/#{output_name}"
+
+    # Trick gclient to reuse our v8 checkouts
+    base = "gclient-cache/chromium.googlesource.com"
+    (buildpath/"#{base}-v8-v8").install Dir["*"]
+
+    depot_tools = resource("depot_tools")
```

Comment:
Extra variable not necessary, makes reading harder (too many depot_tools).
Just inline `resource("depot_tools")` in the two places where it's used.

Reply:
fixed